ntbs-service/Pages/Notifications/Edit/ClinicalDetails.cshtml.cs

```diff
@@ -48,9 +47,14 @@ public override async Task<IActionResult> OnGetAsync(int id, bool isBeingSubmitt
                 return NotFound();
             }
 
-            NotificationBannerModel = new NotificationBannerModel(Notification);
+            await AuthorizeAndSetBannerAsync();
+            if (!HasEditPermission)
+            {
+                return RedirectToOverview(id);
```

Comment:
All the pages redirect to overview when there is no edit permission, which in turn shows a partial view with the warning message instead of the normal content.

For notifications that the user isn't permitted to edit, they will never get to see a direct link to any of the pages apart from overview (which they can click through from search results, for example), so they'd only ever end up here by entering the url directly, which shouldn't really happen. I could of course have shown the same partial view here rather than doing a redirect, however in case we need to do anything more complex in future, I felt it was probably easiest to redirect to overview and let that page handle what to do, what do you think?

Reply:
Good to guard against this, as it's definitely possible that someone will share a link directly to an edit page. 
I think pushing back to the overview is the cleanest way of handling this in a single place.